tests/components/forked_daapd/test_config_flow.py

```diff
@@ -0,0 +1,181 @@
+    data = {
+        CONF_HOST: "192.168.1.1",
+        CONF_PORT: "2345",
+        CONF_PASSWORD: "",
+    }
+    return MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="",
+        data=data,
+        options={},
+        system_options={},
+        source=SOURCE_USER,
+        connection_class=CONN_CLASS_LOCAL_PUSH,
+        entry_id=1,
+    )
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == SOURCE_USER
+
+
+async def test_config_flow(hass, config_entry):
+    """Test that the user step works."""
+    with patch(
+        "homeassistant.components.forked_daapd.config_flow.ForkedDaapdAPI.test_connection"
+    ) as mock_test_connection, patch(
+        "homeassistant.components.forked_daapd.media_player.ForkedDaapdAPI.get_request",
+        autospec=True,
+    ) as mock_get_request:
+        mock_get_request.return_value = SAMPLE_CONFIG
+        mock_test_connection.return_value = ["ok", "My Music on myhost"]
+        config_data = config_entry.data
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=config_data
+        )
+        await hass.async_block_till_done()
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "My Music on myhost"
+        assert result["data"][CONF_HOST] == config_data[CONF_HOST]
+        assert result["data"][CONF_PORT] == config_data[CONF_PORT]
+        assert result["data"][CONF_PASSWORD] == config_data[CONF_PASSWORD]
+
+        # Also test that creating a new entry with the same host aborts
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=config_entry.data,
+        )
+        await hass.async_block_till_done()
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+
+
+async def test_zeroconf_updates_title(hass, config_entry):
+    """Test that zeroconf updates title and aborts with same host."""
+    MockConfigEntry(domain=DOMAIN, data={CONF_HOST: "different host"}).add_to_hass(hass)
+    config_entry.add_to_hass(hass)
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 2
+    discovery_info = {
+        "host": "192.168.1.1",
+        "port": 23,
+        "properties": {"mtd-version": 1, "Machine Name": "zeroconf_test"},
+    }
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}, data=discovery_info
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert config_entry.title == "zeroconf_test"
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)